lua/telescope/health.lua

```diff
@@ -0,0 +1,115 @@
+local fn = vim.fn
+local extension_info = require("telescope").extensions
+
+local health_start = vim.fn["health#report_start"]
+local health_ok = vim.fn["health#report_ok"]
+local health_warn = vim.fn["health#report_warn"]
+local health_error = vim.fn["health#report_error"]
+
+local optional_dependencies = {
+  {
+    finder_name = "live-grep",
+    package = {
+      {
+        name = "rg",
+        url = "[BurntSushi/ripgrep](https://github.com/BurntSushi/ripgrep)",
+        optional = false,
+      },
+    },
+  },
+  {
+    finder_name = "find-files",
+    package = {
+      {
+        name = "fd",
+        url = "[sharkdp/fd](https://github.com/sharkdp/fd)",
+        optional = true,
+      },
+    },
+  },
+}
+
+local required_plugins = {
+  { lib = "popup", optional = false },
+  { lib = "plenary", optional = false },
+  {
+    lib = "nvim-treesitter",
+    optional = true,
+    info = "Install [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) to enable preview highlight capabilities.",
+  },
+}
+
+local check_binary_installed = function(package)
+  local file_extension = jit.os == "Windows" and "exe" or ""
```

Comment:
`jit.os` doesn't return Windows. something like `mingw.....` idk can we not rely on jit. just use `has`